experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1373 @@
+
+parent_type(::Type{DifferencePolyRingElem{T}}) where {T} = DifferencePolyRing{T}
+
+function Base.deepcopy_internal(dpre::DifferencePolyRingElem{T}, dict::IdDict) where {T}
+    # Avoid deepcopying the parent as it may refer back to it in one of its dictionaries 
+    pp = deepcopy_internal(data(dpre), dict)
+    return DifferencePolyRingElem{T}(parent(dpre), pp)
+end
+
+zero(dpr::DifferencePolyRing) = dpr(0)
+
+one(dpr::DifferencePolyRing) = dpr(1)
+
+### Differential ###
+elem_type(::Type{DifferentialPolyRing{T}}) where {T} = DifferentialPolyRingElem{T}
+
+parent_type(::Type{DifferentialPolyRingElem{T}}) where {T} = DifferentialPolyRing{T}
+
+function Base.deepcopy_internal(dpre::DifferentialPolyRingElem{T}, dict::IdDict) where {T}
+    # Avoid deepcopying the parent as it may refer back to it in one of its dictionaries 
+    pp = deepcopy_internal(data(dpre), dict)
+    return DifferentialPolyRingElem{T}(parent(dpre), pp)
+end
+
+zero(dpr::DifferentialPolyRing) = dpr(0)
+
+one(dpr::DifferentialPolyRing) = dpr(1)
+
+### generic ###
+base_ring_type(::Type{<:ActionPolyRing{T}}) where {T} = parent_type(T)
+
+is_square(apre::ActionPolyRingElem) = is_square(data(apre))
+
+Base.sqrt(apre::ActionPolyRingElem; check::Bool = true) = parent(apre)(sqrt(data(apre); check = check))
+
+is_irreducible(apre::ActionPolyRingElem) = is_irreducible(data(apre))
+
+is_unit(apre::ActionPolyRingElem) = is_unit(data(apre))
+
+characteristic(apr::ActionPolyRing) = characteristic(__upr(apr))
+
+###############################################################################
+#
+#  Factorisation
+#
+###############################################################################
+
+function __wrap_factorization_apr(f::Fac{<:UniversalPolyRingElem}, apr::ActionPolyRing)
+   res = Fac{elem_type(apr)}()
+   res.unit = apr(f.unit)
+   for (fact, expo) in f
+      AbstractAlgebra.mulpow!(res, apr(fact), expo)
+   end
+   return res
+end
+
+factor_squarefree(apr::ActionPolyRingElem) = __wrap_factorization_apr(factor_squarefree(data(apr)), parent(apr))
+
+factor(apr::ActionPolyRingElem) = __wrap_factorization_apr(factor(data(apr)), parent(apr))
+
+###############################################################################
+#
+#  Field access and related stuff
+#
+###############################################################################
+
+##### Algebras #####
+
+### Difference ###
+ndiffs(dpr::DifferencePolyRing) = dpr.ndiffs
+
+elementary_symbols(dpr::DifferencePolyRing) = dpr.elementary_symbols
+
+### Differential ###
+ndiffs(dpr::DifferentialPolyRing) = dpr.ndiffs
+
+elementary_symbols(dpr::DifferentialPolyRing) = dpr.elementary_symbols
+
+##### Elements #####
+
+parent(dpre::DifferencePolyRingElem) = dpre.parent
+
+parent(dpre::DifferentialPolyRingElem) = dpre.parent
+
+##### Related stuff #####
+base_ring(apr::ActionPolyRing) = base_ring(__upr(apr))
+
+nelementary_symbols(apr::ActionPolyRing) = length(elementary_symbols(apr))
```

Comment:
```suggestion
n_elementary_symbols(apr::ActionPolyRing) = length(elementary_symbols(apr))
```
this is the name that would be most consistent with the rest of oscar